### PR TITLE
Cache notifications

### PR DESCRIPTION
**Pull request by annando**
We now cache the created notification. This increase the performance of `ping` a lot.

### CODE REVIEW

File: src/Navigation/Notifications/Factory/Notification.php [hunk 1]
```diff
@@ -43,6 +48,8 @@ class Notification extends BaseFactory implements ICanCreateFromTableRow
 	private $l10n;
 	/** @var LocalRelationship */
 	private $localRelationshipRepo;
+	/** @var ICanCache */
+	private $cache;
 
 	public function __construct(\Friendica\App\BaseURL $baseUrl, \Friendica\Core\L10n $l10n, \Friendica\Contact\LocalRelationship\Repository\LocalRelationship $localRelationshipRepo, LoggerInterface $logger)
```
Comment (nupplaphil) [inline src/Navigation/Notifications/Factory/Notification.php]:
```suggestion
	public function __construct(\Friendica\App\BaseURL $baseUrl, \Friendica\Core\L10n $l10n, \Friendica\Contact\LocalRelationship\Repository\LocalRelationship $localRelationshipRepo, LoggerInterface $logger, ICanCache $cache)
```

File: src/Navigation/Notifications/Factory/Notification.php [hunk 2]
```diff
@@ -51,6 +58,7 @@ public function __construct(\Friendica\App\BaseURL $baseUrl, \Friendica\Core\L10
 		$this->baseUrl = $baseUrl;
 		$this->l10n = $l10n;
 		$this->localRelationshipRepo = $localRelationshipRepo;
+		$this->cache = DI::cache(); // @todo Add the correct mechanism for class construction here
```
Comment (nupplaphil) [inline src/Navigation/Notifications/Factory/Notification.php]:
```suggestion
		$this->cache = $cache;
```

File: src/Navigation/Notifications/Factory/Notification.php [hunk 3]
```diff
@@ -107,6 +115,12 @@ public function getMessageFromNotification(Entity\Notification $Notification): a
 	{
 		$message = [];
 
+		$cachekey = 'Notification:' . $Notification->id;
```
Comment (nupplaphil) [inline src/Navigation/Notifications/Factory/Notification.php:116]: Maybe you can use a constant for the prefix instead

File: src/Navigation/Notifications/Factory/Notification.php (hunk 1 above)
Comment (annando) [reply src/Navigation/Notifications/Factory/Notification.php]: I tried this - but then no new notifications had been created. The function is called from several locations and it seems as if it stops working from at least one of the locations when I add the parameter there.

Could this be the problem?
https://github.com/friendica/friendica/blob/2022.05-rc/src/Model/Post/UserNotification.php#L311

File: src/Navigation/Notifications/Factory/Notification.php (hunk 3 above)
Comment (annando) [reply src/Navigation/Notifications/Factory/Notification.php:116]: Since I'm only calling it from one location , I think a constant is not needed here.

**Comment by annando** [comment]
@nupplaphil I found the problem why it hadn't worked with the initialize.

**Pull request merged**